source/extensions/filters/common/ext_authz/ext_authz.h

```diff
@@ -68,16 +68,23 @@ struct Response {
   // A set of HTTP headers returned by the authorization server, that will be optionally appended
   // to the request to the upstream server.
   Http::HeaderVector headers_to_append;
+  // A set of HTTP headers returned by the authorization server, that will be optionally be added
+  // to the request to the upstream server if they don't already exist.
+  Http::HeaderVector headers_to_add_if_absent;
```

Comment:
Since the authorization service can see all of the original request headers (or be configured to so, in the case of an HTTP server), we shouldn't need the conditional "if absent" set since the server can determine whether it exists originally or not. If the header exists, it can just omit it from "headers_to_add" and if it doesn't exist it can be added to "headers_to_add".

For simplicity I think it would be best to drop this particular functionality. What do you think?

Reply:
Thanks, @esmet. Sure I can just remove `headers_to_add_if_absent` from here and also rename `response_headers_to_append` --> `response_headers_to_add` as before.

Reply:
@esmet What action should we take in case a request header comes with `ADD_IF_ABSENT` action? Should we throw an error saying it's not supported for request headers or just append/add it?

Reply:
That's a good point. I realize now that this is coming from a common API (the header value option) and the fact that using it doesn't make much sense for ext_authz (given Check requests), it still makes sense to implement it. Ignoring it or throwing an error would be odd.

I think we can just keep it, then. Sorry for the misguidance.

Reply:
No problem at all. I can revert my changes of removing it and push rest of the changes addressing your comments for a final review. Thank you so much for reviewing.